adapta/utils/README.md

```diff
@@ -107,4 +107,49 @@ with memory_limit(memory_limit_percentage=0.2):
     # Do some stuff here
     # if the provided limit is exceeded, MemoryError will be raised and can be caught by client code
     pass
-```
\ No newline at end of file
+```
+
+### runtime_metrics:
+
+Decorator for logging method execution time and reporting function execution time to metric provider (e.g. DataDog).  
+
+Notes:
+* When using the decorator the wrapped function must take logger and metrics_provider as keyword arguments.
+  * It is recommended to have decorated methods use `**kwargs` if logger and metric providers are not explicitly defined.
+* Logger sends logs with `loglevel.DEBUG`.
+* Run-time metrics are send in seconds. 
+
+Parameters:
+* `metric_name`: Type of method being wrapped. This will be used as a metric_name by the metric provider.
+* `tag_function_name` (optional): If True function name will be added as additional metric_tag. Defaults to False.
+* `metric_tags` (optional): Dictionary containing extra tags used by the metric provider.
+
+Example Usage:
+
+```python
+from adapta.metrics.providers.datadog_provider import DatadogMetricsProvider
+from adapta.logs import SemanticLogger
+from adapta.logs.models import LogLevel
+from adapta.utils import run_time_metrics
+
+@run_time_metrics(metric_name='example')
+def adder(number1, number2, logger=None, **_kwargs):
+    result = number1 + number2
+    logger.info(f'Sum of the numbers {result}')
```

Comment:
```suggestion
    logger.info('Sum of the numbers {result}', result=result)
```
@george-zubrienko dose not like when you f-strings for logging, maybe we should make sure the documentation does not use this discouraged usage.